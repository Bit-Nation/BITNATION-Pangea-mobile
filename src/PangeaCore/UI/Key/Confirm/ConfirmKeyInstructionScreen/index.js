// @flow

import React from 'react';
import {
  View,
  ScrollView,
  Image,
} from 'react-native';

import styles from './styles';
import { androidNavigationButtons, screen } from 'pangea-common-reactnative/Screens';
import BackgroundImage from 'pangea-common-reactnative/UI/BackgroundImage';
import FakeNavigationBar from 'pangea-common-reactnative/UI/FakeNavigationBar';
import PanelView from 'pangea-common-reactnative/UI/PanelView';
import { KEY_LENGTH } from 'pangea-common/Constants';
import AssetsImages from 'pangea-common-reactnative/assets/AssetsImages';
import BodyParagraphs from 'pangea-common-reactnative/UI/BodyParagraphs';
import i18n from 'pangea-common/i18n';
import NavigatorComponent from 'pangea-common-reactnative/UI/NavigatorComponent';
import type { NavigatorProps } from 'pangea-common-reactnative/UI/NavigatorComponent';
import Colors from 'pangea-common-reactnative/styles/colors';

type Props = {
  shouldShowCancel: boolean,
};

class ConfirmKeyInstructionScreen extends NavigatorComponent<Props> {
  static defaultProps;

  constructor(props: Props & NavigatorProps) {
    super(props);

    props.navigator.setButtons({
      leftButtons: props.shouldShowCancel ? [{
        id: 'cancel',
        title: 'Cancel',
        buttonColor: Colors.navigationButtonColor,
      }] : [],
      ...androidNavigationButtons,
    });
  }

  onNavBarButtonPress(id: string) {
    if (id === 'cancel') {
      this.props.navigator.dismissModal();
    }
  }

  onNextButtonPressed = () => {
    if (this.props.navigator) {
      this.props.navigator.push(screen('CONFIRM_KEY_PROCESS_SCREEN'));
    }
  };

  render() {
    return (
      <View style={styles.screenContainer}>
        <BackgroundImage />
        <FakeNavigationBar />

        <View style={styles.bodyContainer}>

          <ScrollView style={styles.scrollView} contentContainerStyle={styles.noflex}>
            <PanelView
              style={styles.panelViewTransparent}
              buttonTitle={i18n.t('screens.confirmKey.startButton')}
              onButtonClick={this.onNextButtonPressed}
            >
              <BodyParagraphs paragraphs={i18n.t('screens.confirmKey.instructions.beforeGrid', { KEY_LENGTH })} />
              <View style={styles.gridContainer}>
                <Image
                  style={styles.privateKeyDemoImage}
                  resizeMode='contain'
                  source={AssetsImages.privateKeyDemo}
                />
              </View>
              <BodyParagraphs paragraphs={i18n.t('screens.confirmKey.instructions.afterGrid')} />

            </PanelView>
          </ScrollView>
        </View>
      </View>
    );
  }
}

ConfirmKeyInstructionScreen.defaultProps = {
  shouldShowCancel: false,
};

export default ConfirmKeyInstructionScreen;
