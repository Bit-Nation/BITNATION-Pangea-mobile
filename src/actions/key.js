// @flow

import { type Mnemonic } from '../types/Mnemonic';

type ValidateEnteredMnemonicAction = { +type: 'VALIDATE_ENTERED_MNEMONIC' };
type ChangeMnemonicValidAction = { +type: 'CHANGE_MNEMONIC_VALID', +mnemonicValid: boolean | null };
type ChangeEnteredMnemonicAction = { +type: 'CHANGE_ENTERED_MNEMONIC', +mnemonic: Mnemonic };

export type Action =
  | ValidateEnteredMnemonicAction
  | ChangeMnemonicValidAction
  | ChangeEnteredMnemonicAction;

export const VALIDATE_ENTERED_MNEMONIC = 'VALIDATE_ENTERED_MNEMONIC';
export const CHANGE_MNEMONIC_VALID = 'CHANGE_MNEMONIC_VALID';
export const CHANGE_ENTERED_MNEMONIC = 'CHANGE_ENTERED_MNEMONIC';

/**
 * @desc Action creator for an action that starts entered mnemonic validation process.
 * @returns {ValidateEnteredMnemonicAction} An action.
 */
export function validateEnteredMnemonic(): ValidateEnteredMnemonicAction {
  return {
    type: VALIDATE_ENTERED_MNEMONIC,
  };
}

/**
 * @desc Action creator for an action that should be called once
 * mnemonic validation status is changed.
 * @param {(boolean|null)} valid Flag that shows if mnemonic is valid.
 * Null is used to mark an undefined state of validation (while it is in progress).
 * @returns {ChangeMnemonicValidAction} An action.
 */
export function changeMnemonicValid(valid: boolean | null): ChangeMnemonicValidAction {
  return {
    type: CHANGE_MNEMONIC_VALID,
    mnemonicValid: valid,
  };
}

/**
 * @desc Action creator for an action that should be called once entered mnemonic is changed.
 * @param {Mnemonic} mnemonic Mnemonic that is entered.
 * @returns {ChangeEnteredMnemonicAction} An action.
 */
export function changeEnteredMnemonic(mnemonic: Mnemonic): ChangeEnteredMnemonicAction {
  return {
    type: CHANGE_ENTERED_MNEMONIC,
    mnemonic,
  };
}
