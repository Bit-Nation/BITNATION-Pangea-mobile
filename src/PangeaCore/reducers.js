// @flow

import { combineReducers } from 'redux';
import accounts, { type State as AccountsState } from '@pangea/accounts/accounts-reducers';
import nations, { type State as NationsState } from '@pangea/nations/nations-reducers';
import modifyNation, { type State as ModifyNationState } from '@pangea/nations/modifyNation-reducers';
import wallet, { type State as WalletState } from '@pangea/wallet/wallet-reducers';
import key, { type State as KeyState } from '@pangea/key/key-reducers';
import activity, { type State as ActivityState } from '@pangea/activity/activity-reducers';
import testingMode, { type State as TestingModeState } from '@pangea/accounts/testingMode-reducers';
import chat, { type State as ChatState } from '@pangea/chat/chat-reducers';
import settings, { type State as SettingsState } from '@pangea/settings/settings-reducers';
import dApps, { type State as DAppsState } from '@pangea/dApps/dApps-reducers';
import documents, { type State as DocumentsState } from '@pangea/documents/documents-reducers';
import contacts, { type State as ContactsState } from '@pangea/contacts/contacts-reducers';

export type State = {
  accounts: AccountsState,
  nations: NationsState,
  modifyNation: ModifyNationState,
  wallet: WalletState,
  key: KeyState,
  activity: ActivityState,
  testingMode: TestingModeState,
  chat: ChatState,
  settings: SettingsState,
  dApps: DAppsState,
  documents: DocumentsState,
  contacts: ContactsState,
}

export const subReducers = {
  accounts,
  nations,
  modifyNation,
  wallet,
  key,
  activity,
  testingMode,
  chat,
  settings,
  dApps,
  documents,
  contacts,
};

export default combineReducers(subReducers);
