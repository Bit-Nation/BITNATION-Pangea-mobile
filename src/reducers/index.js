// @flow

import { combineReducers } from 'redux';
import accounts, { type State as AccountsState } from './accounts';
import nations, { type State as NationsState } from './nations';
import modifyNation, { type State as ModifyNationState } from './modifyNation';
import wallet, { type State as WalletState } from './wallet';
import key, { type State as KeyState } from './key';
import activity, { type State as ActivityState } from './activity';
import testingMode, { type State as TestingModeState } from './testingMode';
import chat, { type State as ChatState } from './chat';
import settings, { type State as SettingsState } from './settings';

export type State = {
  accounts: AccountsState,
  nations: NationsState,
  modifyNation: ModifyNationState,
  wallet: WalletState,
  key: KeyState,
  activity: ActivityState,
  testingMode: TestingModeState,
  chat: ChatState,
  settings: SettingsState,
}

export const subReducers = {
  accounts,
  nations,
  modifyNation,
  wallet,
  key,
  activity,
  testingMode,
  chat,
  settings,
};

export default combineReducers(subReducers);
