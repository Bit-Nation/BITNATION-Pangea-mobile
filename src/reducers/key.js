// @flow

import {
  type Action,
  CHANGE_ENTERED_MNEMONIC,
  CHANGE_MNEMONIC_VALID,
  VALIDATE_ENTERED_MNEMONIC,
} from '../actions/key';
import type { Mnemonic } from '../types/Mnemonic';
import { SERVICES_DESTROYED } from '../actions/serviceContainer';

export type State = {
  +enteredMnemonic: Mnemonic | null,
  +mnemonicValid: boolean | null,
  +mnemonicValidationInProgress: boolean,
}

export const initialState: State = {
  enteredMnemonic: null,
  mnemonicValid: null,
  mnemonicValidationInProgress: false,
};

/**
 * @desc Key reducer.
 * @param {State} state Current state.
 * @param {Action} action Performed action.
 * @returns {State} Next state.
 */
export default (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case SERVICES_DESTROYED:
      return initialState;
    case CHANGE_ENTERED_MNEMONIC:
      return { ...state, enteredMnemonic: action.mnemonic };
    case VALIDATE_ENTERED_MNEMONIC:
      return { ...state, mnemonicValid: null, mnemonicValidationInProgress: true };
    case CHANGE_MNEMONIC_VALID:
      return { ...state, mnemonicValid: action.mnemonicValid, mnemonicValidationInProgress: false };
    default:
      return state;
  }
};

