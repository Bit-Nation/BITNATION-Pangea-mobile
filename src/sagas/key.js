import { all, takeEvery, call, put, select } from 'redux-saga/effects';

import {
  CREATE_PRIVATE_KEY, mnemonicCreated, SAVE_PRIVATE_KEY
} from '../actions/key';
import {
  updateWalletList
} from '../actions/wallet';
import containerPromise from '../services/container';
import { compressMnemonic } from '../utils/key';

async function createPrivateKey() {
  const container = await containerPromise;
  return await container.eth.utils.createPrivateKey();
}

async function privateKeyToMnemonic(privateKey) {
  const container = await containerPromise;
  return await container.eth.utils.privateKeyToMnemonic(privateKey);
}

async function mnemonicToPrivateKey(mnemonic) {
  const container = await containerPromise;
  return await container.eth.utils.mnemonicToPrivateKey(mnemonic);
}

async function savePrivateKey(privateKey) {
  const container = await containerPromise;
  return await container.eth.utils.savePrivateKey(privateKey);
}

function* createPrivateKeySaga() {
  const privateKey = yield call(createPrivateKey);
  const mnemonic = yield call(privateKeyToMnemonic, privateKey);
  yield put(mnemonicCreated(mnemonic));
}

function* watchCreateWallet() {
  yield takeEvery(CREATE_PRIVATE_KEY, createPrivateKeySaga);
}

function* savePrivateKeySaga() {
  const state = yield select();
  const mnemonicArray = state.key.createdMnemonic;
  if (!mnemonicArray) {
    return;
  }
  const mnemonic = compressMnemonic(mnemonicArray);
  const privateKey = yield call(mnemonicToPrivateKey, mnemonic);
  yield call(savePrivateKey, privateKey);

  yield put(updateWalletList());
}

function* watchSavePrivateKey() {
  yield takeEvery(SAVE_PRIVATE_KEY, savePrivateKeySaga);
}

export default function* rootSaga() {
  yield all([
    watchCreateWallet(),
    watchSavePrivateKey(),
  ]);
}

