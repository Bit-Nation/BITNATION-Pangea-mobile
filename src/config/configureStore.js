// @flow

import { createStore, applyMiddleware, compose, type Store } from 'redux';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import createSagaMonitor from '@clarketm/saga-monitor';
import reducers from '../reducers';
import rootSaga from '../sagas';

/**
 * @desc Configures a Redux store.
 * @return {Store} Created store object.
 */
export default function configureStore(): Store {
  const sagaMonitor = createSagaMonitor({
    level: 'log',
    actionDispatch: true,
  });
  const sagaMiddleware = createSagaMiddleware({ sagaMonitor });

  const enhancer = compose(
    applyMiddleware(sagaMiddleware),
    applyMiddleware(logger),
  );

  const store = createStore(reducers, enhancer);
  sagaMiddleware.run(rootSaga);
  return store;
}
