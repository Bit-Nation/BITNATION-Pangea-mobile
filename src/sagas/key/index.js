import { all, takeEvery } from 'redux-saga/effects';

import { CREATE_PRIVATE_KEY, SAVE_PRIVATE_KEY } from '../../actions/key';
import { createPrivateKeySaga, savePrivateKeySaga } from './sagas';


function* watchCreateWallet() {
  yield takeEvery(CREATE_PRIVATE_KEY, createPrivateKeySaga);
}

function* watchSavePrivateKey() {
  yield takeEvery(SAVE_PRIVATE_KEY, savePrivateKeySaga);
}

export default function* rootSaga() {
  yield all([
    watchCreateWallet(),
    watchSavePrivateKey(),
  ]);
}