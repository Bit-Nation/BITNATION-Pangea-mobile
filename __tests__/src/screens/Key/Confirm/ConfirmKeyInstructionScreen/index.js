import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import { initialState } from '../../../../../../src/reducers/key';
import ConfirmKeyInstructionScreen from '../../../../../../src/screens/Key/Confirm/ConfirmKeyInstructionScreen';

test('ConfirmKeyInstructionScreen renders correctly', () => {
  const initialStateMock = {
    key: initialState,
  };
  const storeMock = configureStore([]);
  const propsMock = {
    navigator: {
      push: jest.fn(),
    },
  };

  const wrapper = shallow((
    <ConfirmKeyInstructionScreen {...propsMock} store={storeMock(initialStateMock)} />
  ));
  expect(wrapper).toMatchSnapshot();
});
