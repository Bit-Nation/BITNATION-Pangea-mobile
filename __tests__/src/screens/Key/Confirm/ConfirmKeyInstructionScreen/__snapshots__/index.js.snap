// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmKeyInstructionScreen renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmKeyInstructionScreen
    navigator={
      Object {
        "push": [MockFunction],
        "setOnNavigatorEvent": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
        },
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <BackgroundImage
          source={
            Object {
              "testUri": "../../../src/assets/images/background.jpg",
            }
          }
          style={undefined}
        />,
        <FakeNavigationBar
          navBarHidden={false}
        />,
        <View
          style={
            Object {
              "alignContent": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "marginLeft": 8,
              "marginRight": 8,
            }
          }
        >
          <ScrollViewMock
            contentContainerStyle={
              Object {
                "flex": 0,
              }
            }
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <PanelView
              buttonTitle="Begin"
              onButtonClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 8,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "paddingBottom": 4,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 4,
                }
              }
              titleStyle={
                Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                  "textAlign": "left",
                }
              }
            >
              <BodyParagraphs
                paragraphs={
                  Array [
                    "We will show you a group of 24 words that is the private key of your account.",
                  ]
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  resizeMode="contain"
                  source={1}
                  style={
                    Object {
                      "height": 555.84,
                      "width": 720,
                    }
                  }
                />
              </View>
              <BodyParagraphs
                paragraphs={
                  Array [
                    "Write the words on paper, in order. Store the paper in very safe place. If your device is lost, stolen, broken, or upgraded, you must have this key to restore your account.",
                  ]
                }
              />
            </PanelView>
          </ScrollViewMock>
        </View>,
      ],
      "style": Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "source": Object {
            "testUri": "../../../src/assets/images/background.jpg",
          },
          "style": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "navBarHidden": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ScrollViewMock
            contentContainerStyle={
              Object {
                "flex": 0,
              }
            }
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <PanelView
              buttonTitle="Begin"
              onButtonClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 8,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "paddingBottom": 4,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 4,
                }
              }
              titleStyle={
                Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                  "textAlign": "left",
                }
              }
            >
              <BodyParagraphs
                paragraphs={
                  Array [
                    "We will show you a group of 24 words that is the private key of your account.",
                  ]
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  resizeMode="contain"
                  source={1}
                  style={
                    Object {
                      "height": 555.84,
                      "width": 720,
                    }
                  }
                />
              </View>
              <BodyParagraphs
                paragraphs={
                  Array [
                    "Write the words on paper, in order. Store the paper in very safe place. If your device is lost, stolen, broken, or upgraded, you must have this key to restore your account.",
                  ]
                }
              />
            </PanelView>
          </ScrollViewMock>,
          "style": Object {
            "alignContent": "flex-start",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "marginLeft": 8,
            "marginRight": 8,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <PanelView
              buttonTitle="Begin"
              onButtonClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 8,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "paddingBottom": 4,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 4,
                }
              }
              titleStyle={
                Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                  "textAlign": "left",
                }
              }
            >
              <BodyParagraphs
                paragraphs={
                  Array [
                    "We will show you a group of 24 words that is the private key of your account.",
                  ]
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  resizeMode="contain"
                  source={1}
                  style={
                    Object {
                      "height": 555.84,
                      "width": 720,
                    }
                  }
                />
              </View>
              <BodyParagraphs
                paragraphs={
                  Array [
                    "Write the words on paper, in order. Store the paper in very safe place. If your device is lost, stolen, broken, or upgraded, you must have this key to restore your account.",
                  ]
                }
              />
            </PanelView>,
            "contentContainerStyle": Object {
              "flex": 0,
            },
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "buttonTitle": "Begin",
              "children": Array [
                <BodyParagraphs
                  paragraphs={
                    Array [
                      "We will show you a group of 24 words that is the private key of your account.",
                    ]
                  }
                />,
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={1}
                    style={
                      Object {
                        "height": 555.84,
                        "width": 720,
                      }
                    }
                  />
                </View>,
                <BodyParagraphs
                  paragraphs={
                    Array [
                      "Write the words on paper, in order. Store the paper in very safe place. If your device is lost, stolen, broken, or upgraded, you must have this key to restore your account.",
                    ]
                  }
                />,
              ],
              "onButtonClick": [Function],
              "style": Object {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              },
              "titleStyle": Object {
                "color": "white",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 28,
                "textAlign": "left",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "paragraphs": Array [
                    "We will show you a group of 24 words that is the private key of your account.",
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Image
                    resizeMode="contain"
                    source={1}
                    style={
                      Object {
                        "height": 555.84,
                        "width": 720,
                      }
                    }
                  />,
                  "style": Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "resizeMode": "contain",
                    "source": 1,
                    "style": Object {
                      "height": 555.84,
                      "width": 720,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "paragraphs": Array [
                    "Write the words on paper, in order. Store the paper in very safe place. If your device is lost, stolen, broken, or upgraded, you must have this key to restore your account.",
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <BackgroundImage
            source={
              Object {
                "testUri": "../../../src/assets/images/background.jpg",
              }
            }
            style={undefined}
          />,
          <FakeNavigationBar
            navBarHidden={false}
          />,
          <View
            style={
              Object {
                "alignContent": "flex-start",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "marginLeft": 8,
                "marginRight": 8,
              }
            }
          >
            <ScrollViewMock
              contentContainerStyle={
                Object {
                  "flex": 0,
                }
              }
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <PanelView
                buttonTitle="Begin"
                onButtonClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 8,
                    "flexDirection": "column",
                    "justifyContent": "space-between",
                    "marginBottom": 8,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 8,
                    "paddingBottom": 4,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 4,
                  }
                }
                titleStyle={
                  Object {
                    "color": "white",
                    "fontFamily": "Roboto",
                    "fontSize": 22,
                    "fontWeight": "bold",
                    "letterSpacing": 0,
                    "lineHeight": 28,
                    "textAlign": "left",
                  }
                }
              >
                <BodyParagraphs
                  paragraphs={
                    Array [
                      "We will show you a group of 24 words that is the private key of your account.",
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={1}
                    style={
                      Object {
                        "height": 555.84,
                        "width": 720,
                      }
                    }
                  />
                </View>
                <BodyParagraphs
                  paragraphs={
                    Array [
                      "Write the words on paper, in order. Store the paper in very safe place. If your device is lost, stolen, broken, or upgraded, you must have this key to restore your account.",
                    ]
                  }
                />
              </PanelView>
            </ScrollViewMock>
          </View>,
        ],
        "style": Object {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "source": Object {
              "testUri": "../../../src/assets/images/background.jpg",
            },
            "style": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "navBarHidden": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <ScrollViewMock
              contentContainerStyle={
                Object {
                  "flex": 0,
                }
              }
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <PanelView
                buttonTitle="Begin"
                onButtonClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 8,
                    "flexDirection": "column",
                    "justifyContent": "space-between",
                    "marginBottom": 8,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 8,
                    "paddingBottom": 4,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 4,
                  }
                }
                titleStyle={
                  Object {
                    "color": "white",
                    "fontFamily": "Roboto",
                    "fontSize": 22,
                    "fontWeight": "bold",
                    "letterSpacing": 0,
                    "lineHeight": 28,
                    "textAlign": "left",
                  }
                }
              >
                <BodyParagraphs
                  paragraphs={
                    Array [
                      "We will show you a group of 24 words that is the private key of your account.",
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={1}
                    style={
                      Object {
                        "height": 555.84,
                        "width": 720,
                      }
                    }
                  />
                </View>
                <BodyParagraphs
                  paragraphs={
                    Array [
                      "Write the words on paper, in order. Store the paper in very safe place. If your device is lost, stolen, broken, or upgraded, you must have this key to restore your account.",
                    ]
                  }
                />
              </PanelView>
            </ScrollViewMock>,
            "style": Object {
              "alignContent": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "marginLeft": 8,
              "marginRight": 8,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <PanelView
                buttonTitle="Begin"
                onButtonClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 8,
                    "flexDirection": "column",
                    "justifyContent": "space-between",
                    "marginBottom": 8,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 8,
                    "paddingBottom": 4,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 4,
                  }
                }
                titleStyle={
                  Object {
                    "color": "white",
                    "fontFamily": "Roboto",
                    "fontSize": 22,
                    "fontWeight": "bold",
                    "letterSpacing": 0,
                    "lineHeight": 28,
                    "textAlign": "left",
                  }
                }
              >
                <BodyParagraphs
                  paragraphs={
                    Array [
                      "We will show you a group of 24 words that is the private key of your account.",
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={1}
                    style={
                      Object {
                        "height": 555.84,
                        "width": 720,
                      }
                    }
                  />
                </View>
                <BodyParagraphs
                  paragraphs={
                    Array [
                      "Write the words on paper, in order. Store the paper in very safe place. If your device is lost, stolen, broken, or upgraded, you must have this key to restore your account.",
                    ]
                  }
                />
              </PanelView>,
              "contentContainerStyle": Object {
                "flex": 0,
              },
              "style": Object {
                "flex": 1,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "buttonTitle": "Begin",
                "children": Array [
                  <BodyParagraphs
                    paragraphs={
                      Array [
                        "We will show you a group of 24 words that is the private key of your account.",
                      ]
                    }
                  />,
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Image
                      resizeMode="contain"
                      source={1}
                      style={
                        Object {
                          "height": 555.84,
                          "width": 720,
                        }
                      }
                    />
                  </View>,
                  <BodyParagraphs
                    paragraphs={
                      Array [
                        "Write the words on paper, in order. Store the paper in very safe place. If your device is lost, stolen, broken, or upgraded, you must have this key to restore your account.",
                      ]
                    }
                  />,
                ],
                "onButtonClick": [Function],
                "style": Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 8,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "paddingBottom": 4,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 4,
                },
                "titleStyle": Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                  "textAlign": "left",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "paragraphs": Array [
                      "We will show you a group of 24 words that is the private key of your account.",
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Image
                      resizeMode="contain"
                      source={1}
                      style={
                        Object {
                          "height": 555.84,
                          "width": 720,
                        }
                      }
                    />,
                    "style": Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "resizeMode": "contain",
                      "source": 1,
                      "style": Object {
                        "height": 555.84,
                        "width": 720,
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "paragraphs": Array [
                      "Write the words on paper, in order. Store the paper in very safe place. If your device is lost, stolen, broken, or upgraded, you must have this key to restore your account.",
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
